### PR TITLE
Added basic CRUD for RIAK using Django

### PR DESCRIPTION
**Pull request by Akshay14469**

### CODE REVIEW

File: NHS.Api/README.md [hunk 1]
```diff
@@ -7,12 +7,12 @@ INSTALLATION :-
 Run requirement27.txt file.
 
 API's :-
-Search : "http://<Domain Name>:<port_number>/<bucket_name>/search/<key_name>/"
-Write : "http://<Domain Name>:<port_number>/<bucket_name>/write/<key_name>/"
-Delete : "http://<Domain Name>:<port_number>/<bucket_name>/delete/<key_name>/"
-Update : "http://<Domain Name>:<port_number>/<bucket_name>/update/<key_name>/"
+Search : "http://<Domain Name>:<port_number>/search/<key_name>/"
+Write : "http://<Domain Name>:<port_number>/write/<key_name>/"
```
Comment (prabhat14176) [inline NHS.Api/README.md]: It should have to be a post method

File: NHS.Api/README.md [hunk 2]
```diff
@@ -7,12 +7,12 @@ INSTALLATION :-
 Run requirement27.txt file.
 
 API's :-
-Search : "http://<Domain Name>:<port_number>/<bucket_name>/search/<key_name>/"
-Write : "http://<Domain Name>:<port_number>/<bucket_name>/write/<key_name>/"
-Delete : "http://<Domain Name>:<port_number>/<bucket_name>/delete/<key_name>/"
-Update : "http://<Domain Name>:<port_number>/<bucket_name>/update/<key_name>/"
+Search : "http://<Domain Name>:<port_number>/search/<key_name>/"
```
Comment (prabhat14176) [inline NHS.Api/README.md]: with this name the endpoint seems to be so generics ,
try to make it specific as search_trust or search_org , 

do the same for all

File: NHS.Api/README.md [hunk 3]
```diff
@@ -7,12 +7,12 @@ INSTALLATION :-
...
+Write : "http://<Domain Name>:<port_number>/write/<key_name>/"
+Delete : "http://<Domain Name>:<port_number>/delete/<key_name>/"
+Update : "http://<Domain Name>:<port_number>/update/<key_name>/"
```
Comment (prabhat14176) [inline NHS.Api/README.md]: Either comment it or remove. for this POC better if we will comment it.

File: NHS.Api/riak_crud/urls.py [hunk 4]
```diff
@@ -2,8 +2,8 @@
 from . import views as riak_views
 
 urlpatterns = [
-    url("^(?P<bucket>[a-zA-Z]+)/search/(?P<key>[a-zA-Z0-8]+)$", riak_views.search, name="search"),
-    url("^(?P<bucket>[a-zA-Z]+)/write/(?P<key>[a-zA-Z0-8]+)$", riak_views.write, name="write"),
-    url("^(?P<bucket>[a-zA-Z]+)/update/(?P<key>[a-zA-Z0-8]+)$", riak_views.update, name="update"),
-    url("^(?P<bucket>[a-zA-Z]+)/delete/(?P<key>[a-zA-Z0-8]+)$", riak_views.delete, name="delete"),
+    url("^search/(?P<key>[a-zA-Z0-8]+)$", riak_views.search, name="search"),
+    url("^write/(?P<key>[a-zA-Z0-8]+)$", riak_views.write, name="write"),
+    url("^update/(?P<key>[a-zA-Z0-8]+)$", riak_views.update, name="update"),
```
Comment (prabhat14176) [inline NHS.Api/riak_crud/urls.py]: try to remove or comment it

File: NHS.Api/riak_crud/views.py [hunk 5]
```diff
@@ -38,26 +38,30 @@
 }
 
 
-def search(request, bucket, key):
-    bucket = riak_crud.riak_client.bucket(bucket)
+def search(request, key):
+    bucket = riak_crud.riak_client.bucket(riak_crud.bucket_name)
```
Comment (prabhat14176) [inline NHS.Api/riak_crud/views.py]: use 
bucket = _get_or_create_bucket()
instead 
bucket = riak_crud.riak_client.bucket(riak_crud.bucket_name)

File: NHS.Api/riak_crud/views.py [hunk 6]
```diff
@@ -38,26 +38,30 @@
...
 
 
-def delete(request, bucket, key):
-    bucket = riak_crud.riak_client.bucket(bucket)
+def _get_or_create_bucket():
+    return riak_crud.riak_client.bucket(riak_crud.bucket_name)
+
+
+def delete(request, key):
+    bucket = riak_crud.riak_client.bucket(riak_crud.bucket_name)
```
Comment (prabhat14176) [inline NHS.Api/riak_crud/views.py]: use
bucket = _get_or_create_bucket()
instead
bucket = riak_crud.riak_client.bucket(riak_crud.bucket_name)

File: NHS.Api/riak_crud/urls.py [hunk 7]
```diff
@@ -2,8 +2,8 @@
...
+    url("^delete/(?P<key>[a-zA-Z0-8]+)$", riak_views.delete, name="delete"),
```
Comment (prabhat14176) [inline NHS.Api/riak_crud/urls.py]: Try to give specific name to these views , 

Ex : search - search_trust or search_org

File: NHS.Api/riak_crud/views.py (hunk 5 above)
Comment (prabhat14176) [inline NHS.Api/riak_crud/views.py]: use _get_or_create_bucket()

File: NHS.Api/riak_crud/views.py [hunk 8]
```diff
@@ -38,26 +38,30 @@
...
+    bucket = _get_or_create_bucket()
     trust_data = bucket.new(key, data=DATA)
     trust_data.store()
     return HttpResponse("Write Success")
 
 
-def delete(request, bucket, key):
-    bucket = riak_crud.riak_client.bucket(bucket)
+def _get_or_create_bucket():
+    return riak_crud.riak_client.bucket(riak_crud.bucket_name)
```
Comment (prabhat14176) [inline NHS.Api/riak_crud/views.py]: _get_or_create_bucket()

File: NHS.Api/manage.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
+import os
+import sys
+
+if __name__ == "__main__":
+    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "riak27.settings")
+    try:
+        from django.core.management import execute_from_command_line
+    except ImportError:
+        # The above import may fail for some other reason. Ensure that the
```
Comment (prabhat14176) [inline NHS.Api/manage.py]: remove commented code

File: NHS.Api/riak_crud/__init__.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import riak
+
+
+def create_client():
+    # client = riak.RiakClient(host="18.217.190.19")
+    client = riak.RiakClient(host="18.188.197.228")
```
Comment (vijendr13545) [inline NHS.Api/riak_crud/__init__.py]: host ip should be stored in config.

File: NHS.Api/requirement27.txt [hunk 11]
```diff
@@ -0,0 +1,6 @@
+basho-erlastic==2.1.1
```
Comment (prabhat14176) [inline NHS.Api/requirement27.txt:1]: As we moved dependency to setup , so if this file is not required, remove or make a comment on top of it

File: NHS.Api/riak27/urls.py [hunk 12]
```diff
@@ -0,0 +1,22 @@
+"""riak27 URL Configuration
+
+The `urlpatterns` list routes URLs to views. For more information please see:
```
Comment (prabhat14176) [inline NHS.Api/riak27/urls.py]: remove commented code

File: NHS.Api/riak27/settings.py [hunk 13]
```diff
@@ -0,0 +1,121 @@
+"""
+Django settings for riak27 project.
```
Comment (prabhat14176) [inline NHS.Api/riak27/settings.py]: remove commented code

File: NHS.Api/riak_crud/__init__.py [hunk 14]
```diff
@@ -0,0 +1,10 @@
+import riak
+
+
+def create_client():
+    # client = riak.RiakClient(host="18.217.190.19")
```
Comment (prabhat14176) [inline NHS.Api/riak_crud/__init__.py]: remove commented code

File: NHS.Api/riak_crud/urls.py [hunk 15]
```diff
@@ -0,0 +1,9 @@
+from django.conf.urls import url
+from . import views as riak_views
+
+urlpatterns = [
+    url("^(?P<bucket>[a-zA-Z]+)/search/(?P<key>[a-zA-Z0-8]+)$", riak_views.search, name="search"),
```
Comment (prabhat14176) [inline NHS.Api/riak_crud/urls.py]: views name seem to be generics

File: NHS.Api/riak_crud/views.py [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+
+from django.http import JsonResponse, HttpResponse
+import riak_crud
+
+# This Hardcoded data is to be removed.
+# Schema of DATA is not final yet, It might change.
+DATA = {
+    "E1": {
+        "avg_gap": 10,
+        "patient_count": 100,
```
Comment (prabhat14176) [inline NHS.Api/riak_crud/views.py]: as discussed,  Apply same json format

**Review by vijendr13545 — COMMENTED** [review]
Overall looks good to me but I added a comment. Please have a look.

**Review by prabhat14176 — COMMENTED** [review]
Try to resolve the comments, 
rest seems good to me

File: NHS.Api/requirement27.txt (hunk 11 above)
Comment (Akshay14469) [reply NHS.Api/requirement27.txt:1]: I have kept requirement.txt file as a backup type, if in any case setup.py fails then requirement.txt file will serve the purpose.

File: NHS.Api/README.md [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+# Write : "http://<Domain Name>:<port_number>/write_region/<region_key_name>/"
+# Delete : "http://<Domain Name>:<port_number>/delete_region/<region_key_name>/"
+# Update : "http://<Domain Name>:<port_number>/update_region/<region_key_name>/"
+
+**Note :- Update API is not functional till now. Key_name(Name of Trust / Name of Region).
+
+Mock Data :
+The schema of Mock data is temporary and can be changed. For time being I have hard coded the data to be written in RIAK.
+
+TrustData = {
```
Comment (prabhat14176) [inline NHS.Api/README.md:26]: We are storing the gap between the events,

So it should be 
TrustData = {
    "E1": 20, (Difference in days between E1 and E2 start time ) 
    "E2": 100,
    "E3": 25,
    "E4": 32
}
same will be followed for rest and for RegionData  as well

File: NHS.Api/README.md [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+    "E1": 20,
+    "E2": 30,
+    "E3": 40,
+    "E4": 50,
+    "E5": 60
+}
+
+
+RIAK SERVER CONFIG :-
+Riak server config is present inside the riak_crud application. For time being the host is present at "18.217.190.19"
```
Comment (prabhat14176) [inline NHS.Api/README.md]: Try to replace this IP from QA IP address

File: NHS.Api/riak27/settings.py [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = '2e3kucnc$=&nzq0#4d@ryuv5e6b^sb!zh7x66h!01v(%iy8nmh'
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
+
+ALLOWED_HOSTS = ["*"]
```
Comment (prabhat14176) [inline NHS.Api/riak27/settings.py]: Ideally it should point to localhost,

File: NHS.Api/riak27/settings.py [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+
+RIAK_HOST_IP = "3.15.181.1"
```
Comment (prabhat14176) [inline NHS.Api/riak27/settings.py]: replace from QA Ip

File: NHS.Api/riak_crud/views.py [hunk 21]
```diff
@@ -0,0 +1,124 @@
...
+from __future__ import unicode_literals
+
+from django.http import JsonResponse, HttpResponse
+from django.views.decorators.http import require_POST
+import riak_crud
+
+# This Hardcoded data is to be removed.
+# Schema of DATA is not final yet, It might change.
+# Repetition of CRUD Logic is to be removed.
+TrustData = {
```
Comment (prabhat14176) [inline NHS.Api/riak_crud/views.py:12]: Try to refer the format mentioned in earlier comments

File: NHS.Api/riak_crud/views.py [hunk 22]
```diff
@@ -0,0 +1,124 @@
...
+    return HttpResponse("Region Delete Success")
+
+
+# @require_POST
+def update_region(request, region_key):
+    # TODO Add Update Functionality
+    return HttpResponse("Region Update Success")
+
+
+def _get_or_create_trust_bucket():
```
Comment (prabhat14176) [inline NHS.Api/riak_crud/views.py]: Try to make it generic as "_get_or_create_bucket( **bucket_name**)"

File: NHS.Api/riak_crud/views.py [hunk 23]
```diff
@@ -0,0 +1,124 @@
...
+    return riak_crud.riak_client.bucket(riak_crud.trust_bucket_name)
+
+
+def _get_or_create_region_bucket():
+    return riak_crud.riak_client.bucket(riak_crud.region_bucket_name)
```
Comment (prabhat14176) [inline NHS.Api/riak_crud/views.py]: Try to move this key also to setting.py

File: NHS.Api/setup.py [hunk 24]
```diff
@@ -0,0 +1,9 @@
+import setuptools
+
+setuptools.setup(
+    name="Riak2.7",
+    author="Akshay",
```
Comment (prabhat14176) [inline NHS.Api/setup.py]: Replace name "Akshay" with NHSD

**Review by prabhat14176 — COMMENTED** [review]
Do the changes as suggested in review comments

**Review by prabhat14176 — APPROVED** [review]
I am merging this to master.

**Pull request merged**